geth/jail/console/console.go

```diff
@@ -0,0 +1,130 @@
+package console
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/node"
+)
+
+const (
+	// EventConsoleLog defines the event type for the console.log call.
+	EventConsoleLog = "vm.console.log"
+
+	// EventConsoleWarn defines the event type for the console.debug call.
+	EventConsoleWarn = "vm.console.warn"
+
+	// EventConsoleDebug defines the event type for the console.debug call.
+	EventConsoleDebug = "vm.console.debug"
+
+	// EventConsoleError defines the event type for the console.error call.
+	EventConsoleError = "vm.console.error"
+)
+
+// Extension takes a giving extension function and writer and returns a standard otto.Function
+// callable by a otto vm.
+func Extension(w io.Writer, ext func(otto.FunctionCall, io.Writer) otto.Value) func(otto.FunctionCall) otto.Value {
+	return func(fn otto.FunctionCall) otto.Value {
+		return ext(fn, w)
+	}
+}
+
+// Log provides the function caller for handling console.log
+// calls as replacement for the default console.log function within a
+// otto.Otto VM instance.
+func Log(fn otto.FunctionCall, w io.Writer) otto.Value {
```

Comment:
`Log`, `Warn`, `Error` and `Debug` look very much alike. What do you think about creating a helper private function like this:
```go
func log(fn otto.FunctionCall, w io.Writer, type string) otto.Value {
  node.SendSignal(node.SignalEnvelope{
    Type:  type,
    Event: convertArgs(fn.ArgumentList),
  })

  handleConsole(w, "%s: %s", type, fn.ArgumentList)

  return otto.UndefinedValue()
}
```
Then all these functions I mentioned at the beginning could be one-liners.